core/src/solve.jl

```diff
@@ -479,15 +487,17 @@ end
 
 function set_current_basin_properties!(
     basin::Basin,
-    storage::AbstractVector{Float64},
-    t::Real,
+    current_area,
+    current_level,
```

Comment:
What is the reason for not taking the `current_area` and `current_level` out of the already passed `basin` argument? Same for `formulate!` directly below.

Reply:
This is again about me not knowing much about `get_tmp`, and whether it is an expensive call.